frontend/lib/resolution/can-pass.cpp

```diff
@@ -828,6 +828,12 @@ CanPassResult CanPassResult::canInstantiate(Context* context,
         return instantiate();
       }
     }
+  } else if (auto actualPt = actualT->toCPtrType()) {
+    if (auto formalPt = formalT->toCPtrType()) {
```

Comment:
This should also handle implicit conversions around `c_ptr_void` (or whatever the new form of it is, `c_ptr(void)`? ). AFAIK we can implicitly convert any `c_ptr` to the void pointer.